presto-main/src/main/java/com/facebook/presto/catalogserver/CatalogServer.java

```diff
@@ -29,132 +29,242 @@
 import com.facebook.presto.transaction.TransactionManager;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
 
 import javax.inject.Inject;
 
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Optional;
+import java.util.OptionalLong;
 
 import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
 
 @ThriftService(value = "presto-catalog-server", idlName = "PrestoCatalogServer")
 public class CatalogServer
 {
     private static final String EMPTY_STRING = "";
```

Comment:
Would this be the ideal way of doing it? Because I still would need to pass an empty value to the returnValue. Could pass null instead but im not sure if that will cause any issues

>        return viewsList.isEmpty()
>               ? new MetadataEntry<>(EMPTY_STRING, isCacheHit)
>               : new MetadataEntry<>(writeValueAsString(viewsList, objectMapper), isCacheHit);

>        return viewsList.isEmpty()
>               ? new MetadataEntry<>((null or EMPTY_STRING), isCacheHit, true)
>               : new MetadataEntry<>(writeValueAsString(viewsList, objectMapper), isCacheHit, false);